core/src/text/textBuffer.cpp

```diff
@@ -91,15 +96,28 @@ bool TextBuffer::addLabel(const TextStyle::Parameters& _params, Label::Transform
     uint32_t stroke = (_params.strokeColor & 0x00ffffff) + (strokeWidth << 24);
 
     for (auto& q : quads) {
+        // the atlas resolution at which the quad uvs have been generated onto
+        glm::vec2 atlasResolution(q.atlasWidth, q.atlasHeight);
+        glm::vec2 uvScaleFactor(1.f, 1.f);
+
+        if (atlasResolution != m_atlasRes) {
+            uvScaleFactor = atlasResolution / m_atlasRes;
+        }
```

Comment:
I went the straight way since it's a blocker for eraser-map content, and I think this could be merged with non-fixed size UV first. I will create a task for it, and follow up on it. But yeah I definitely agree that it will help us for packing vertex data.